### PR TITLE
Remove CTranslatorPlStmtToDXL Deadcode

### PR DESCRIPTION
**Pull request by vraghavan78**
Cleaning dead code. [#119102697]

Context: When we did not have optimization modules in Orca and just DXL, we used CTranslatorPlStmtToDXL to test correctness of translation of DXL to PlStmt.

@danielgustafsson @oarap @d @xinzweb @hsyuan please take a look.

### CODE REVIEW

File: src/backend/gpopt/utils/funcs.cpp [hunk 1]
```diff
@@ -50,8 +50,6 @@ PG_FUNCTION_INFO_V1(DumpPlan);
 PG_FUNCTION_INFO_V1(RestorePlan);
 PG_FUNCTION_INFO_V1(DumpPlanToFile);
 PG_FUNCTION_INFO_V1(RestorePlanFromFile);
-PG_FUNCTION_INFO_V1(DumpPlanDXL);
```
Comment (hsyuan) [inline src/backend/gpopt/utils/funcs.cpp]: https://github.com/greenplum-db/gporca/blob/master/scripts/log_runner/setup.sql
in orca, there is one function `run_planfreezer_test ` that is using DumpPlanDXL, make sure to delete that piece of code in GPORCA too.

File: src/backend/gpopt/translate/Makefile [hunk 2]
```diff
@@ -34,6 +34,5 @@ OBJS =  CMappingColIdVar.o \
 		CTranslatorDXLToQuery.o \
 		CTranslatorQueryToDXL.o \
 		CTranslatorDXLToPlStmt.o \
```
Comment (yuanhsh) [inline src/backend/gpopt/translate/Makefile]: delete " \" at the end

Comment (hsyuan) [inline src/backend/gpopt/translate/Makefile]: delete " \" at the end

**Comment by hsyuan** [comment]
- In GPORCA, make sure you remove two functions (DumpPlanDXL & DumpPlanToDXLFile) in load.sql.in:
https://github.com/greenplum-db/gporca/blob/master/scripts/load.sql.in

- In GPDB, make sure you also update load.sql:
https://github.com/greenplum-db/gpdb/blob/master/src/backend/gpopt/utils/load.sql
In fact, I think this file can be deleted safely, because it is generated by the template file load.sql.in.

File: src/backend/gpopt/utils/funcs.cpp (hunk 1 above)
Comment (vraghavan78) [inline src/backend/gpopt/utils/funcs.cpp]: will do.

**Comment by vraghavan78** [comment]
Good point.

**Comment by hsyuan** [comment]
OK, no problem. I can open a separate PR to remove that file, since it is apparently pointing to someone's local absolute path, which doesn't make any sense.

**Comment by danielgustafsson** [comment]
Looks good to me, :+1: on getting this in. In the mergebranch I have pushed a temporary commit doing mostly this which was required in order to get it to compile: https://github.com/danielgustafsson/gpdb/commit/fe01e981bed7a02e8b8b1f08cf048b7d5eee9d1e

**Comment by oarap** [comment]
+1

**Pull request closed without merging**